spec/models/revenue_estimation_spec.rb

```diff
@@ -48,4 +48,34 @@
       end
     end
   end
+
+  describe '#calculate_quantity' do
+    context 'with valid inputs' do
+      it 'calculates quantity based on revenue and average_ticket' do
```

Comment:
Necessary but not sufficient condition to turn RevenueEstimation model invalid. We do not want RevenueEstimation to be created with invalid params, or nil params. Intead of checking nil, `revenue_estimation.valid?` must be false. It will prevent its creation and the `root_path` will not raise error since `revenue_estimation.quantity` is used on the dasboard.

You might check [numericality of the inputs with Rails helpers](https://guides.rubyonrails.org/active_record_validations.html#numericality).

Another good way to prove this validation is using [Shoulda Matcher gem](https://github.com/thoughtbot/shoulda-matchers/blob/main/lib/shoulda/matchers/active_model/validate_numericality_of_matcher.rb). It is already set in the project. It will increase your productivity creating tests.

Reply:
Great idea! I just adjusted it